cmd/clusterctl/client/tree/discovery.go

```diff
@@ -69,9 +69,9 @@ func Discovery(ctx context.Context, c client.Client, namespace, name string, opt
 	}
 
```

Comment:
I'd add some validation and error messages in here, rather that in `external.Get`. The error coming from `external.Get` doesn't have enough context for the user to fix, while here we know which object we're working with.

Reply:
@JosephSalisbury THoughts about this?